feature/selector.py

```diff
@@ -511,6 +515,78 @@ def benchmark(selectors: Dict[str, Union[SelectionMethod.Correlation,
     check_true(selectors is not None, ValueError("Benchmark selectors cannot be none."))
     check_true(data is not None, ValueError("Benchmark data cannot be none."))
 
+    if cv is None:
+        return _bench(selectors=selectors,
+                      data=data,
+                      labels=labels,
+                      output_filename=output_filename,
+                      drop_zero_variance_features=drop_zero_variance_features,
+                      verbose=verbose)
+    else:
+
+        # Create K-Fold object
+        kf = KFold(n_splits=cv, shuffle=True, random_state=Constants.default_seed)
+
+        # Initialize variables
+        t0 = time()
+        train_labels, test_labels = None, None
+        score_df, selected_df, runtime_df = pd.DataFrame(), pd.DataFrame(), pd.DataFrame()
+
+        # Split data into cv-folds and run _bench for each fold
+        if verbose:
+            print("\n>>> Running")
+        for fold, (train_index, _) in enumerate(kf.split(data)):
+
+            if verbose:
+                print("\tFold", fold, "...")
+
+            # Split data, labels into folds
+            train_data = data.iloc[train_index]
+            if labels is not None:
+                train_labels = labels.iloc[train_index]
+
+            # Run benchmark
+            score_cv_df, selected_cv_df, runtime_cv_df = _bench(selectors=selectors,
+                                                                data=train_data,
+                                                                labels=train_labels,
+                                                                output_filename=output_filename,
+                                                                drop_zero_variance_features=drop_zero_variance_features,
+                                                                verbose=False)
+
+            # Concatenate data frames
```

Comment:
I am trying to understand the overall change in the input - output spec. So help me here.

When CV is None, we have score_df, selected_df, runtime_df as before, and when we have folds, now we have the same score_df, selected_df, runtime_df but there are rows appended for each fold? 

If I were to run the benchmark(cv=5) and want to see what happened on fold=3, I need to calculate a certain index in the output dfs? 

Also, does it mean the indexes will be repeated for each feature after concat?  (I assume this is how are able to group_by in calc_stats()?) 

===

And the input/output of calc_stat() has not changed from the previous setup, right? It is just that, the score_mean now is the mean of "the mean of" the folds? 

=== 

So we have an implicit connection between bench() and calc_stats() that is based on "concatenation"?

Reply:
* BTW, I am perfectly okay if you tell me "getting the results of fold =3 alone does not make sense, so that case should not 
occur") 

* Analogy, recwiser returns a list of results when the CV is used. So basically same output format, just multiple times in a list for every fold. MABWiser on predict(context) returns an arm, predict(contexts) returns a List of arms. I am totally okay if we have different behavior here, but just checking if "coherence" makes sense 
*

Reply:
I think your understanding is correct. 

I did consider adding a new 'fold' field to make easier to identify the results from a specific fold as you suggested. However, I can't really see a reason for needing to this. Generally users will be interested in calculating aggregated statistics across the folds (like we do in `calculate_statistics`).

It is still possible for the astute user to get the results for the fold=3 by selecting the following slice of rows: 
(3-1)*num_variables:(3)*num_variables (but again, it's not obvious to me why this would be needed)

Reply:
Agreed.. I don't see why anyone would want fold = 3, and if there is such person they should be able to figure it out. Calculate stats is what matters more there.

Reply:
+1 here as well. The only reason you'd want to see fold = 3 is for diagnostics, and this architecture still allows for that.